frontend/libs/studio-components/src/components/StudioTextResourcePicker/StudioTextResourcePicker.tsx

```diff
@@ -4,36 +4,49 @@ import type { TextResource } from '../../types/TextResource';
 import type { StudioComboboxProps } from '../StudioCombobox';
 import { StudioCombobox } from '../StudioCombobox';
 import type { Override } from '../../types/Override';
+import classes from './StudioTextResourcePicker.module.css';
 
 export type StudioTextResourcePickerProps = Override<
   {
-    emptyListText: string;
-    onValueChange: (id: string) => void;
+    onValueChange: (id: string | null) => void;
     textResources: TextResource[];
+    unsetOptionLabel: string;
     value?: string;
   },
   StudioComboboxProps
 >;
 
 export const StudioTextResourcePicker = forwardRef<HTMLInputElement, StudioTextResourcePickerProps>(
-  ({ textResources, onSelect, onValueChange, emptyListText, value, ...rest }, ref) => {
-    const handleValueChange = useCallback(([id]: string[]) => onValueChange(id), [onValueChange]);
+  ({ textResources, onSelect, onValueChange, unsetOptionLabel, value, ...rest }, ref) => {
+    const handleValueChange = useCallback(
+      ([id]: string[]) => onValueChange(id || null),
+      [onValueChange],
+    );
 
     return (
       <StudioCombobox
         hideLabel
         onValueChange={handleValueChange}
-        value={value ? [value] : []}
+        value={value ? [value] : ['']}
         {...rest}
         ref={ref}
       >
-        <StudioCombobox.Empty>{emptyListText}</StudioCombobox.Empty>
+        {renderUnsetOption(unsetOptionLabel)}
         {renderTextResourceOptions(textResources)}
       </StudioCombobox>
     );
   },
 );
 
+function renderUnsetOption(label: string): ReactElement {
+  // This cannot be a component function since the option component must be a direct child of the combobox component.
```

Comment:
Do we need this comment?

Reply:
This is a case where we need to deviate from general best practices because of limitations in an external library (The Design System). A developer that looks at this in the future might spend a lot of time trying to find out why it doesn't work making this a component, which would be the default approach. Therefore this comment.